master/internal/kubernetes/pod.go

```diff
@@ -101,9 +153,191 @@ func (p *pod) startPod(ctx *actor.Context) error {
 	}
 }
 
-func (p *pod) configureResourcesRequirements() v1.ResourceRequirements {
-	return v1.ResourceRequirements{
-		Limits: map[v1.ResourceName]resource.Quantity{
+func (p *pod) receivePodStatusUpdate(ctx *actor.Context, msg podStatusUpdate) error {
+	p.pod = msg.updatedPod
+
+	switch msg.updatedPod.Status.Phase {
+	case k8sV1.PodPending:
+		// When pods are deleted, Kubernetes sometimes transitions pod statuses
+		// to pending prior to deleting them. We ignore this state if we have
+		// triggered the deletion.
+		if p.resourcesDeleted {
+			ctx.Log().WithField("pod", p.podName).Info(
+				"ignoring pod status because resources are being deleted")
```

Comment:
```suggestion
				"ignoring pod status because resources have been deleted")
```

Reply:
imo are been is a bit more accurate, because if we are getting a `PodPending` status, that means k8 is in the process of deleting the pod.